dev/Dockerfile

```diff
@@ -128,13 +128,27 @@ COPY --from=builder /app/target/release/typegate /bin/typegate
 COPY --from=builder /app/typegate/deno.jsonc /app/typegate/import_map.json ./typegate/
 COPY --from=builder /app/typegate/engine/*.js /app/typegate/engine/*.ts ./typegate/engine/
 COPY --from=builder /app/typegate/src ./typegate/src/
+COPY --from=builder /app/typegate/tests ./typegate/tests/
 COPY dev/LICENSE-Elastic-2.0.md LICENSE.md
 
 # writeable
 COPY --from=builder --chown=nonroot:nonroot /deno-dir /deno-dir
 COPY --from=builder --chown=nonroot:nonroot /app/typegate/deno.lock ./typegate/
 COPY --from=builder --chown=nonroot:nonroot /app/typegate/tmp ./
 
+# distroless-nonroot is by default the nonroot user
+# which prevents us from removing the /deno bin down below
+USER root
+
+# we temporarliy copy the deno bin and run the type checker
+# to make sure all required ts files are found in the image
+COPY --from=deno-bin --chown=nonroot:nonroot /usr/bin/deno /bin/deno
```

Comment:
This likely will have an effect on the image size, maybe running that operation in another target in the docker file would save us that?

Reply:
Oof, forgot about that. Using a separate `check` target does seem to help! Down to 347 megs from 488. Surprised it's 373 tbh. The `typegate` bin is apprently 192 megs now. (and the fat version of the `meta-cli` is 194 :/). I guess 347 makes sense when you account for the 80 megabyte `libwasmedge.so`, all the pre-downloaded `ts` files...etc.

Reply:
Wait a minute, `meta-cli` fat is only 70 megs gzipped. The 347 number's for the metatype image as a whole. Oci does gzip the tars, right?

Reply:
Does it only compress the images in transit? Because the deno binary making a 150 megabyte difference only makes sense if the image layer tars are stored on disk uncompressed.

Reply:
Websearch seems to confirm this. I suppose having to decompress large tarballs everytime an local image is run is not exactly efficient.

Reply:
@Yohe-Am yeah, we would need to take a different approach (we shall test the copy and paste of every binary) or try with https://github.com/slimtoolkit/slim - but not urgent, 300Mo uncompressed is fine